<robot name="ur10e_robot">
<link name="base_link"/>
<link name="base_link_inertia">
    <visual>
        <origin rpy="0.0 0.0 3.141592653589793" xyz="0.0 0.0 0.0"/>
        <geometry>
            <mesh filename="package://bullet_ros_ur/meshes/ur10e/visual/base.dae"/>
        </geometry>
        <material name="LightGrey">
            <color rgba="0.7 0.7 0.7 1.0"/>
        </material>
    </visual>
    <collision>
        <origin rpy="0.0 0.0 3.141592653589793" xyz="0.0 0.0 0.0"/>
        <geometry>
            <mesh filename="package://UR10e_meshes/meshes/ur10e/collision/base_vhacd.obj"/>
        </geometry>
    </collision>
    <inertial>
        <mass value="4.0"/>
        <origin rpy="0.0 0.0 0.0" xyz="0.0 0.0 0.0"/>
        <inertia ixx="0.0061063308908" ixy="0.0" ixz="0.0" iyy="0.0061063308908" iyz="0.0" izz="0.01125"/>
    </inertial>
</link>
<link name="shoulder_link">
    <visual>
        <origin rpy="0.0 0.0 3.141592653589793" xyz="0.0 0.0 0.0"/>
        <geometry>
            <mesh filename="package://bullet_ros_ur/meshes/ur10e/visual/shoulder.dae"/>
        </geometry>
        <material name="LightGrey">
            <color rgba="0.7 0.7 0.7 1.0"/>
        </material>
    </visual>
    <collision>
        <origin rpy="0.0 0.0 3.141592653589793" xyz="0.0 0.0 0.0"/>
        <geometry>
            <mesh filename="package://UR10e_meshes/meshes/ur10e/collision/shoulder_vhacd.obj"/>
        </geometry>
    </collision>
    <inertial>
        <mass value="7.778"/>
        <origin rpy="0.0 0.0 0.0" xyz="0.0 0.0 0.0"/>
        <inertia ixx="0.03147431257693659" ixy="0.0" ixz="0.0" iyy="0.03147431257693659" iyz="0.0" izz="0.021875624999999996"/>
    </inertial>
</link>
<link name="upper_arm_link">
    <visual>
        <origin rpy="1.5707963267948966 0.0 -1.5707963267948966" xyz="0.0 0.0 0.1762"/>
        <geometry>
            <mesh filename="package://bullet_ros_ur/meshes/ur10e/visual/upperarm.dae"/>
        </geometry>
        <material name="LightGrey">
            <color rgba="0.7 0.7 0.7 1.0"/>
        </material>
    </visual>
    <collision>
        <origin rpy="1.5707963267948966 0.0 -1.5707963267948966" xyz="0.0 0.0 0.1762"/>
        <geometry>
            <mesh filename="package://UR10e_meshes/meshes/ur10e/collision/upperarm_vhacd.obj"/>
        </geometry>
    </collision>
    <inertial>
        <mass value="12.93"/>
        <origin rpy="0.0 1.5707963267948966 0.0" xyz="-0.306 0.0 0.175"/>
        <inertia ixx="0.42175380379841093" ixy="0.0" ixz="0.0" iyy="0.42175380379841093" iyz="0.0" izz="0.03636562499999999"/>
    </inertial>
</link>
<link name="forearm_link">
    <visual>
        <origin rpy="1.5707963267948966 0.0 -1.5707963267948966" xyz="0.0 0.0 0.0393"/>
        <geometry>
            <mesh filename="package://bullet_ros_ur/meshes/ur10e/visual/forearm.dae"/>
        </geometry>
        <material name="LightGrey">
            <color rgba="0.7 0.7 0.7 1.0"/>
        </material>
    </visual>
    <collision>
        <origin rpy="1.5707963267948966 0.0 -1.5707963267948966" xyz="0.0 0.0 0.0393"/>
        <geometry>
            <mesh filename="package://UR10e_meshes/meshes/ur10e/collision/forearm_vhacd.obj"/>
        </geometry>
    </collision>
    <inertial>
        <mass value="3.87"/>
        <origin rpy="0.0 1.5707963267948966 0.0" xyz="-0.285775 0.0 0.0393"/>
        <inertia ixx="0.11079302548902206" ixy="0.0" ixz="0.0" iyy="0.11079302548902206" iyz="0.0" izz="0.010884375"/>
    </inertial>
</link>
<link name="wrist_1_link">
    <visual>
        <origin rpy="1.5707963267948966 0.0 0.0" xyz="0.0 0.0 -0.135"/>
        <geometry>
            <mesh filename="package://bullet_ros_ur/meshes/ur10e/visual/wrist1.dae"/>
        </geometry>
        <material name="LightGrey">
            <color rgba="0.7 0.7 0.7 1.0"/>
        </material>
    </visual>
    <collision>
        <origin rpy="1.5707963267948966 0.0 0.0" xyz="0.0 0.0 -0.135"/>
        <geometry>
            <mesh filename="package://UR10e_meshes/meshes/ur10e/collision/wrist1_vhacd.obj"/>
        </geometry>
    </collision>
    <inertial>
        <mass value="1.96"/>
        <origin rpy="0.0 0.0 0.0" xyz="0.0 0.0 0.0"/>
        <inertia ixx="0.005108247956699999" ixy="0.0" ixz="0.0" iyy="0.005108247956699999" iyz="0.0" izz="0.005512499999999999"/>
    </inertial>
</link>
<link name="wrist_2_link">
    <visual>
        <origin rpy="0.0 0.0 0.0" xyz="0.0 0.0 -0.12"/>
        <geometry>
            <mesh filename="package://bullet_ros_ur/meshes/ur10e/visual/wrist2.dae"/>
        </geometry>
        <material name="LightGrey">
            <color rgba="0.7 0.7 0.7 1.0"/>
        </material>
    </visual>
    <collision>
        <origin rpy="0.0 0.0 0.0" xyz="0.0 0.0 -0.12"/>
        <geometry>
            <mesh filename="package://UR10e_meshes/meshes/ur10e/collision/wrist2_vhacd.obj"/>
        </geometry>
    </collision>
    <inertial>
        <mass value="1.96"/>
        <origin rpy="0.0 0.0 0.0" xyz="0.0 0.0 0.0"/>
        <inertia ixx="0.005108247956699999" ixy="0.0" ixz="0.0" iyy="0.005108247956699999" iyz="0.0" izz="0.005512499999999999"/>
    </inertial>
</link>
<link name="wrist_3_link">
    <visual>
        <origin rpy="1.5707963267948966 0.0 0.0" xyz="0.0 0.0 -0.1168"/>
        <geometry>
            <mesh filename="package://bullet_ros_ur/meshes/ur10e/visual/wrist3.dae"/>
        </geometry>
        <material name="LightGrey">
            <color rgba="0.7 0.7 0.7 1.0"/>
        </material>
    </visual>
    <collision>
        <origin rpy="1.5707963267948966 0.0 0.0" xyz="0.0 0.0 -0.1168"/>
        <geometry>
            <mesh filename="package://UR10e_meshes/meshes/ur10e/collision/wrist3_vhacd.obj"/>
        </geometry>
    </collision>
    <inertial>
        <mass value="0.202"/>
        <origin rpy="0.0 0.0 0.0" xyz="0.0 0.0 -0.025"/>
        <inertia ixx="0.00014434577559500002" ixy="0.0" ixz="0.0" iyy="0.00014434577559500002" iyz="0.0" izz="0.00020452500000000002"/>
    </inertial>
</link>
<link name="base">
    <inertial>
        <mass value="0.0"/>
        <inertia ixx="0.0" ixy="0.0" ixz="0.0" iyy="0.0" iyz="0.0" izz="0.0"/>
    </inertial>
</link>
<link name="flange">
    <inertial>
        <mass value="0.0"/>
        <inertia ixx="0.0" ixy="0.0" ixz="0.0" iyy="0.0" iyz="0.0" izz="0.0"/>
    </inertial>
</link>
<link name="tool0">
    <inertial>
        <mass value="0.0"/>
        <inertia ixx="0.0" ixy="0.0" ixz="0.0" iyy="0.0" iyz="0.0" izz="0.0"/>
    </inertial>
</link>
<link name="onrobot_rg6_base_link">
    <inertial>
        <origin rpy="0.0 0.0 0.0" xyz="0.0 0.0 0.0"/>
        <mass value="0.7"/>
        <inertia ixx="0.001" ixy="1e-06" ixz="1e-06" iyy="0.001" iyz="1e-06" izz="0.001"/>
    </inertial>
    <visual>
        <origin rpy="0.0 0.0 0.0" xyz="0.0 0.0 0.0"/>
        <geometry>
            <mesh filename="package://bullet_ros_ur/meshes/rg6/visual/base_link.stl"/>
        </geometry>
        <material name="">
            <color rgba="0.8 0.8 0.8 1.0"/>
        </material>
    </visual>
    <collision>
        <origin rpy="0.0 0.0 0.0" xyz="0.0 0.0 0.0"/>
        <geometry>
            <mesh filename="package://UR10e_meshes/meshes/rg6/collision/base_link_vhacd.obj"/>
        </geometry>
    </collision>
</link>
<link name="jaws_link">
    <inertial>
        <mass value="0.0"/>
        <inertia ixx="0.0" ixy="0.0" ixz="0.0" iyy="0.0" iyz="0.0" izz="0.0"/>
    </inertial>
</link>
<link name="gripper_link">
    <inertial>
        <mass value="0.0"/>
        <inertia ixx="0.0" ixy="0.0" ixz="0.0" iyy="0.0" iyz="0.0" izz="0.0"/>
    </inertial>
</link>
<link name="left_outer_knuckle">
    <inertial>
        <origin rpy="0.0 0.0 0.0" xyz="0.0 0.0 0.0"/>
        <mass value="0.05"/>
        <inertia ixx="0.001" ixy="1e-06" ixz="1e-06" iyy="0.001" iyz="1e-06" izz="0.001"/>
    </inertial>
    <visual>
        <origin rpy="0.0 0.0 0.0" xyz="0.0 0.0 0.0"/>
        <geometry>
            <mesh filename="package://bullet_ros_ur/meshes/rg6/visual/outer_knuckle.stl"/>
        </geometry>
        <material name="">
            <color rgba="0.8 0.8 0.8 1.0"/>
        </material>
    </visual>
    <collision>
        <origin rpy="0.0 0.0 0.0" xyz="0.0 0.0 0.0"/>
        <geometry>
            <mesh filename="package://UR10e_meshes/meshes/rg6/collision/left_outer_knuckle_vhacd.obj"/>
        </geometry>
    </collision>
</link>
<link name="left_inner_knuckle">
    <inertial>
        <origin rpy="0.0 0.0 0.0" xyz="0.0 0.0 0.0"/>
        <mass value="0.05"/>
        <inertia ixx="0.001" ixy="1e-06" ixz="1e-06" iyy="0.001" iyz="1e-06" izz="0.001"/>
    </inertial>
    <visual>
        <origin rpy="0.0 0.0 0.0" xyz="0.0 0.0 0.0"/>
        <geometry>
            <mesh filename="package://bullet_ros_ur/meshes/rg6/visual/inner_knuckle.stl"/>
        </geometry>
        <material name="">
            <color rgba="0.8 0.8 0.8 1.0"/>
        </material>
    </visual>
    <collision>
        <origin rpy="0.0 0.0 0.0" xyz="0.0 0.0 0.0"/>
        <geometry>
            <mesh filename="package://UR10e_meshes/meshes/rg6/collision/left_inner_knuckle_vhacd.obj"/>
        </geometry>
    </collision>
</link>
<link name="left_inner_finger">
    <inertial>
        <origin rpy="0.0 0.0 0.0" xyz="0.0 0.0 0.0"/>
        <mass value="0.05"/>
        <inertia ixx="0.001" ixy="1e-06" ixz="1e-06" iyy="0.001" iyz="1e-06" izz="0.001"/>
    </inertial>
    <visual>
        <origin rpy="0.0 0.0 0.0" xyz="0.0 0.0 0.0"/>
        <geometry>
            <mesh filename="package://bullet_ros_ur/meshes/rg6/visual/inner_finger.stl"/>
        </geometry>
        <material name="">
            <color rgba="0.1 0.1 0.1 1.0"/>
        </material>
    </visual>
    <collision>
        <origin rpy="0.0 0.0 0.0" xyz="0.0 0.0 0.0"/>
        <geometry>
            <mesh filename="package://UR10e_meshes/meshes/rg6/collision/left_inner_finger_vhacd.obj"/>
        </geometry>
    </collision>
</link>
<link name="right_outer_knuckle">
    <inertial>
        <origin rpy="0.0 0.0 0.0" xyz="0.0 0.0 0.0"/>
        <mass value="0.05"/>
        <inertia ixx="0.001" ixy="1e-06" ixz="1e-06" iyy="0.001" iyz="1e-06" izz="0.001"/>
    </inertial>
    <visual>
        <origin rpy="0.0 0.0 0.0" xyz="0.0 0.0 0.0"/>
        <geometry>
            <mesh filename="package://bullet_ros_ur/meshes/rg6/visual/outer_knuckle.stl"/>
        </geometry>
        <material name="">
            <color rgba="0.8 0.8 0.8 1.0"/>
        </material>
    </visual>
    <collision>
        <origin rpy="0.0 0.0 0.0" xyz="0.0 0.0 0.0"/>
        <geometry>
            <mesh filename="package://UR10e_meshes/meshes/rg6/collision/outer_knuckle_vhacd.obj"/>
        </geometry>
    </collision>
</link>
<link name="right_inner_knuckle">
    <inertial>
        <origin rpy="0.0 0.0 0.0" xyz="0.0 0.0 0.0"/>
        <mass value="0.05"/>
        <inertia ixx="0.001" ixy="1e-06" ixz="1e-06" iyy="0.001" iyz="1e-06" izz="0.001"/>
    </inertial>
    <visual>
        <origin rpy="0.0 0.0 0.0" xyz="0.0 0.0 0.0"/>
        <geometry>
            <mesh filename="package://bullet_ros_ur/meshes/rg6/visual/inner_knuckle.stl"/>
        </geometry>
        <material name="">
            <color rgba="0.8 0.8 0.8 1.0"/>
        </material>
    </visual>
    <collision>
        <origin rpy="0.0 0.0 0.0" xyz="0.0 0.0 0.0"/>
        <geometry>
            <mesh filename="package://UR10e_meshes/meshes/rg6/collision/inner_knuckle_vhacd.obj"/>
        </geometry>
    </collision>
</link>
<link name="right_inner_finger">
    <inertial>
        <origin rpy="0.0 0.0 0.0" xyz="0.0 0.0 0.0"/>
        <mass value="0.05"/>
        <inertia ixx="0.001" ixy="1e-06" ixz="1e-06" iyy="0.001" iyz="1e-06" izz="0.001"/>
    </inertial>
    <visual>
        <origin rpy="0.0 0.0 0.0" xyz="0.0 0.0 0.0"/>
        <geometry>
            <mesh filename="package://bullet_ros_ur/meshes/rg6/visual/inner_finger.stl"/>
        </geometry>
        <material name="">
            <color rgba="0.1 0.1 0.1 1.0"/>
        </material>
    </visual>
    <collision>
        <origin rpy="0.0 0.0 0.0" xyz="0.0 0.0 0.0"/>
        <geometry>
            <mesh filename="package://UR10e_meshes/meshes/rg6/collision/inner_finger_vhacd.obj"/>
        </geometry>
    </collision>
</link>
<link name="airskin_0">
    <inertial>
        <mass value="0.0"/>
        <inertia ixx="0.0" ixy="0.0" ixz="0.0" iyy="0.0" iyz="0.0" izz="0.0"/>
    </inertial>
</link>
<link name="airskin_1">
    <inertial>
        <mass value="0.0"/>
        <inertia ixx="0.0" ixy="0.0" ixz="0.0" iyy="0.0" iyz="0.0" izz="0.0"/>
    </inertial>
</link>
<link name="airskin_2">
    <inertial>
        <mass value="0.0"/>
        <inertia ixx="0.0" ixy="0.0" ixz="0.0" iyy="0.0" iyz="0.0" izz="0.0"/>
    </inertial>
</link>
<link name="airskin_3">
    <inertial>
        <mass value="0.0"/>
        <inertia ixx="0.0" ixy="0.0" ixz="0.0" iyy="0.0" iyz="0.0" izz="0.0"/>
    </inertial>
</link>
<link name="airskin_4">
    <inertial>
        <mass value="0.0"/>
        <inertia ixx="0.0" ixy="0.0" ixz="0.0" iyy="0.0" iyz="0.0" izz="0.0"/>
    </inertial>
</link>
<link name="airskin_5">
    <inertial>
        <mass value="0.0"/>
        <inertia ixx="0.0" ixy="0.0" ixz="0.0" iyy="0.0" iyz="0.0" izz="0.0"/>
    </inertial>
</link>
<link name="airskin_6">
    <inertial>
        <mass value="0.0"/>
        <inertia ixx="0.0" ixy="0.0" ixz="0.0" iyy="0.0" iyz="0.0" izz="0.0"/>
    </inertial>
</link>
<link name="airskin_7">
    <inertial>
        <mass value="0.0"/>
        <inertia ixx="0.0" ixy="0.0" ixz="0.0" iyy="0.0" iyz="0.0" izz="0.0"/>
    </inertial>
</link>
<link name="airskin_8">
    <inertial>
        <mass value="0.0"/>
        <inertia ixx="0.0" ixy="0.0" ixz="0.0" iyy="0.0" iyz="0.0" izz="0.0"/>
    </inertial>
</link>
<link name="airskin_9">
    <inertial>
        <mass value="0.0"/>
        <inertia ixx="0.0" ixy="0.0" ixz="0.0" iyy="0.0" iyz="0.0" izz="0.0"/>
    </inertial>
</link>
<link name="airskin_10">
    <inertial>
        <mass value="0.0"/>
        <inertia ixx="0.0" ixy="0.0" ixz="0.0" iyy="0.0" iyz="0.0" izz="0.0"/>
    </inertial>
</link>
<link name="real_world">
    <inertial>
        <mass value="0.0"/>
        <inertia ixx="0.0" ixy="0.0" ixz="0.0" iyy="0.0" iyz="0.0" izz="0.0"/>
    </inertial>
</link>
<link name="table1">
    <inertial>
        <mass value="10.0"/>
        <inertia ixx="1.25" ixy="0.0" ixz="0.0" iyy="2.24" iyz="0.0" izz="2.2"/>
    </inertial>
    <visual>
        <geometry>
            <mesh filename="package://UR10e_meshes/meshes/table/visual/table.obj"/>
        </geometry>
        <material name="Orange">
            <color rgba="1.0 0.7 0.0 1.0"/>
        </material>
        <origin rpy="0.0 0.0 0.0" xyz="0.0 0.0 -0.05"/>
    </visual>
    <collision>
        <geometry>
            <mesh filename="package://UR10e_meshes/meshes/table/collision/table_vhacd.obj"/>
        </geometry>
        <origin rpy="0.0 0.0 0.0" xyz="0.0 0.0 -0.05"/>
    </collision>
</link>
<link name="table1_mount">
    <inertial>
        <mass value="0.0"/>
        <inertia ixx="0.0" ixy="0.0" ixz="0.0" iyy="0.0" iyz="0.0" izz="0.0"/>
    </inertial>
</link>
<joint name="base_link_base_link_inertia" type="fixed">
    <origin rpy="0.0 0.0 3.141592653589793" xyz="0.0 0.0 0.0"/>
    <parent link="base_link"/>
    <child link="base_link_inertia"/>
</joint>
<joint name="shoulder_pan_joint" type="revolute">
    <origin rpy="0.0 0.0 -1.90097651550647e-07" xyz="0.0 0.0 0.1809877580908094"/>
    <axis xyz="0.0 0.0 1.0"/>
    <limit effort="330.0" lower="-6.283185307179586" upper="6.283185307179586" velocity="2.2863813201125716"/>
    <dynamics damping="0.0" friction="0.0"/>
    <parent link="base_link_inertia"/>
    <child link="shoulder_link"/>
</joint>
<joint name="shoulder_lift_joint" type="revolute">
    <origin rpy="1.570460766672098 0.0 -3.944537106944846e-06" xyz="0.0001654711577649769 0.0 0.0"/>
    <axis xyz="0.0 0.0 1.0"/>
    <limit effort="330.0" lower="-6.283185307179586" upper="6.283185307179586" velocity="2.2863813201125716"/>
    <dynamics damping="0.0" friction="0.0"/>
    <parent link="shoulder_link"/>
    <child link="upper_arm_link"/>
</joint>
<joint name="elbow_joint" type="revolute">
    <origin rpy="3.1398871213204 -3.14132430956149 -3.141592613259764" xyz="-0.6123211101113366 0.0 0.0"/>
    <axis xyz="0.0 0.0 1.0"/>
    <limit effort="150.0" lower="-2.6179938779914944" upper="2.6179938779914944" velocity="3.3335788713091694"/>
    <dynamics damping="0.0" friction="0.0"/>
    <parent link="upper_arm_link"/>
    <child link="forearm_link"/>
</joint>
<joint name="wrist_1_joint" type="revolute">
    <origin rpy="3.138153975097802 3.141169145914082 -3.141586795892695" xyz="-0.5711103346125506 0.0006004070349949782 0.1746033163278061"/>
    <axis xyz="0.0 0.0 1.0"/>
    <limit effort="56.0" lower="-3.7524578917878086" upper="0.6108652381980153" velocity="3.3335788713091694"/>
    <dynamics damping="0.0" friction="0.0"/>
    <parent link="forearm_link"/>
    <child link="wrist_1_link"/>
</joint>
<joint name="wrist_2_joint" type="revolute">
    <origin rpy="1.570945953502502 0.0 -2.84695771717025e-06" xyz="0.0001127593189565299 -0.1197786351506083 -1.792208295275806e-05"/>
    <axis xyz="0.0 0.0 1.0"/>
    <limit effort="56.0" lower="-2.6179938779914944" upper="2.705260340591211" velocity="3.3335788713091694"/>
    <dynamics damping="0.0" friction="0.0"/>
    <parent link="wrist_1_link"/>
    <child link="wrist_2_link"/>
</joint>
<joint name="wrist_3_joint" type="revolute">
    <origin rpy="1.569792652461999 3.141592653589793 3.141592621107618" xyz="7.250097681839555e-05 0.115643181850237 -0.0001160681323719041"/>
    <axis xyz="0.0 0.0 1.0"/>
    <limit effort="56.0" lower="-6.283185307179586" upper="5.235987755982989" velocity="3.3335788713091694"/>
    <dynamics damping="0.0" friction="0.0"/>
    <parent link="wrist_2_link"/>
    <child link="wrist_3_link"/>
</joint>
<joint name="base_link_base_fixed_joint" type="fixed">
    <origin rpy="0.0 0.0 3.141592653589793" xyz="0.0 0.0 0.0"/>
    <parent link="base_link"/>
    <child link="base"/>
</joint>
<joint name="wrist_3_flange" type="fixed">
    <origin rpy="0.0 -1.5707963267948966 -1.5707963267948966" xyz="0.0 0.0 0.0"/>
    <parent link="wrist_3_link"/>
    <child link="flange"/>
</joint>
<joint name="flange_tool0" type="fixed">
    <origin rpy="1.5707963267948966 0.0 1.5707963267948966" xyz="0.0 0.0 0.0"/>
    <parent link="flange"/>
    <child link="tool0"/>
</joint>
<joint name="rg6_ur10_joint" type="fixed">
    <origin rpy="0.0 1.57 0.0" xyz="0.0 0.0 0.0"/>
    <parent link="flange"/>
    <child link="onrobot_rg6_base_link"/>
</joint>
<joint name="jaws_joint" type="prismatic">
    <origin rpy="0.0 0.0 0.0" xyz="0.0 0.0 0.0"/>
    <axis xyz="0.0 0.0 1.0"/>
    <limit effort="1000.0" lower="0.0" upper="10.0" velocity="2.0"/>
    <parent link="tool0"/>
    <child link="jaws_link"/>
</joint>
<joint name="gripper_joint" type="fixed">
    <origin rpy="0 0.0 1.57" xyz="0.0 0.0 0.1667"/>
    <parent link="onrobot_rg6_base_link"/>
    <child link="gripper_link"/>
</joint>
<joint name="finger_joint" type="revolute">
    <origin rpy="0.0 0.0 0" xyz="0.0 -0.024112 0.136813"/>
    <axis xyz="-1.0 0.0 0.0"/>
    <limit effort="1000.0" lower="-0.628319" upper="0.628319" velocity="2.0"/>
    <parent link="onrobot_rg6_base_link"/>
    <child link="left_outer_knuckle"/>
</joint>
<joint name="left_inner_knuckle_joint" type="revolute">
    <origin rpy="0.0 0.0 0.0" xyz="0.0 -0.01272 0.1595"/>
    <axis xyz="1.0 0.0 0.0"/>
    <limit effort="1000.0" lower="-0.628319" upper="0.628319" velocity="2.0"/>
    <mimic joint="finger_joint" multiplier="-1.0" offset="0.0"/>
    <parent link="onrobot_rg6_base_link"/>
    <child link="left_inner_knuckle"/>
</joint>
<joint name="left_inner_finger_joint" type="revolute">
    <origin rpy="0.0 0.0 0.0" xyz="0.0 -0.047334999999999995 0.064495"/>
    <axis xyz="1.0 0.0 0.0"/>
    <limit effort="1000.0" lower="-0.872665" upper="0.872665" velocity="2.0"/>
    <mimic joint="finger_joint" multiplier="1.0" offset="0.0"/>
    <parent link="left_outer_knuckle"/>
    <child link="left_inner_finger"/>
</joint>
<joint name="right_outer_knuckle_joint" type="revolute">
    <origin rpy="0.0 0.0 3.141592653589793" xyz="0.0 0.024112 0.136813"/>
    <axis xyz="1.0 0.0 0.0"/>
    <limit effort="1000.0" lower="-0.628319" upper="0.628319" velocity="2.0"/>
    <mimic joint="finger_joint" multiplier="-1.0" offset="0.0"/>
    <parent link="onrobot_rg6_base_link"/>
    <child link="right_outer_knuckle"/>
</joint>
<joint name="right_inner_knuckle_joint" type="revolute">
    <origin rpy="0.0 0.0 -3.141592653589793" xyz="0.0 0.01272 0.1595"/>
    <axis xyz="1.0 0.0 0.0"/>
    <limit effort="1000.0" lower="-0.628319" upper="0.628319" velocity="2.0"/>
    <mimic joint="finger_joint" multiplier="-1.0" offset="0.0"/>
    <parent link="onrobot_rg6_base_link"/>
    <child link="right_inner_knuckle"/>
</joint>
<joint name="right_inner_finger_joint" type="revolute">
    <origin rpy="0.0 0.0 0.0" xyz="0.0 -0.047334999999999995 0.064495"/>
    <axis xyz="1.0 0.0 0.0"/>
    <limit effort="1000.0" lower="-0.872665" upper="0.872665" velocity="2.0"/>
    <mimic joint="finger_joint" multiplier="1.0" offset="0.0"/>
    <parent link="right_outer_knuckle"/>
    <child link="right_inner_finger"/>
</joint>
<joint name="airskin_0_joint" type="fixed">
    <origin rpy="0.0 0.0 0.0" xyz="0.0 0.0 0"/>
    <parent link="upper_arm_link"/>
    <child link="airskin_0"/>
</joint>
<joint name="airskin_1_joint" type="fixed">
    <origin rpy="-1.57 0 0" xyz="-0.325 0.0 0.175"/>
    <parent link="upper_arm_link"/>
    <child link="airskin_1"/>
</joint>
<joint name="airskin_2_joint" type="fixed">
    <origin rpy="1.57 0 0" xyz="-0.325 0 0.175"/>
    <parent link="upper_arm_link"/>
    <child link="airskin_2"/>
</joint>
<joint name="airskin_3_joint" type="fixed">
    <origin rpy="0.0 0.0 0.0" xyz="-0.625 0.0 0.175"/>
    <parent link="upper_arm_link"/>
    <child link="airskin_3"/>
</joint>
<joint name="airskin_4_joint" type="fixed">
    <origin rpy="0.0 3.14 0.0" xyz="-0.01 0.0 0.04"/>
    <parent link="forearm_link"/>
    <child link="airskin_4"/>
</joint>
<joint name="airskin_5_joint" type="fixed">
    <origin rpy="-1.57 0 0.0" xyz="-0.325 0.0 0.04"/>
    <parent link="forearm_link"/>
    <child link="airskin_5"/>
</joint>
<joint name="airskin_6_joint" type="fixed">
    <origin rpy="1.57 0.0 0.0" xyz="-0.325 -0.0 0.04"/>
    <parent link="forearm_link"/>
    <child link="airskin_6"/>
</joint>
<joint name="airskin_7_joint" type="fixed">
    <origin rpy="0.0 3.14 0.0" xyz="-0.575 0.0 0.04"/>
    <parent link="forearm_link"/>
    <child link="airskin_7"/>
</joint>
<joint name="airskin_8_joint" type="fixed">
    <origin rpy="-1.57 0 0" xyz="0.0 0.0 0.0"/>
    <parent link="wrist_1_link"/>
    <child link="airskin_8"/>
</joint>
<joint name="airskin_9_joint" type="fixed">
    <origin rpy="1.57 0.0 0.0" xyz="0.0 0 0.0"/>
    <parent link="wrist_2_link"/>
    <child link="airskin_9"/>
</joint>
<joint name="airskin_10_joint" type="fixed">
    <origin rpy="0 1.57 0" xyz="0.0 0 0.1"/>
    <parent link="onrobot_rg6_base_link"/>
    <child link="airskin_10"/>
</joint>

<joint name="real_world_base_joint" type="fixed">
    <origin rpy="0.0 0.0 1.57" xyz="0.0 0.0 0.0"/>
    <parent link="base_link"/>
    <child link="real_world"/>
</joint>
<joint name="table1_joint" type="fixed">
    <origin rpy="0.0 0.0 0.0" xyz="0.54 0.0 0.0025"/>
    <parent link="real_world"/>
    <child link="table1"/>
</joint>
<joint name="table1_mount_joint" type="fixed">
    <origin rpy="0.0 0.0 0.0" xyz="-0.4625 0.0 0.0"/>
    <parent link="table1"/>
    <child link="table1_mount"/>
</joint>
</robot>